detectors/resources/src/main/resources/META-INF/services/io.opentelemetry.sdk.resources.ResourceProvider

```diff
@@ -1 +1,2 @@
-com.google.cloud.opentelemetry.detectors.GCEResource
\ No newline at end of file
+com.google.cloud.opentelemetry.detectors.GCEResource
+com.google.cloud.opentelemetry.detectors.GKEResource
```

Comment:
How do these resource detects work?  Is it "last one wins"? Is it "all of them win"?

TL;DR: how do we ensure GKE beats GCE when it needs to?

Reply:
1. It's "last write wins" in the current implementation: https://github.com/open-telemetry/opentelemetry-java/blob/bace5dce1f042d76ebbfec3e888f136361d58556/sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java#L182-L183
2. But I'm not sure if the order in the `META-INF` guarantees that classes are loaded in the same order (or if it's deterministic at all)? They are loaded using `ServiceLoader`: https://github.com/open-telemetry/opentelemetry-java/blob/bace5dce1f042d76ebbfec3e888f136361d58556/sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java#L105-L109

For now it shouldn't matter which one wins I think, but if we later add support for `cloud.infrastructure_service` (see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/cloud.md#cloud), we will need for `gcp_gke` to win over `gcp_gce`.

Reply:
Is the SPI going to be frozen after 1.0? I.e., will we be able to add a prioritization mechanism or will we have to couple/merge the GCE and GKE detectors to accomplish this goal?

Reply:
Not he SPI is not freezing with 1.0 yet, confirmed in the last insturmentation SIG.  I'm going to be making some contributions there, but I think we MAY want to iron out a test to make sure GCE/GKE detectors don't conflict.

@clamclamyan can you just open an Issue for us to figure this out later?  We don't need to hold up the PR on this, I was just curious how you were testing this locally and not running into issues.